smartsim/_core/generation/generator.py

```diff
@@ -240,119 +269,60 @@ def _execute_commands(cls, cmd_list: CommandList) -> None:
 
     @staticmethod
     def _mkdir_file(file_path: pathlib.Path) -> Command:
+        """Build a Command to create a directory, including any
+        necessary parent directories.
+
+        :param file_path: The directory path to be created
+        :return: A Command object to execute the directory creation
+        """
         cmd = Command(["mkdir", "-p", str(file_path)])
         return cmd
 
     @staticmethod
     def _copy_files(
-        files: t.Union[EntityFiles, None], dest: pathlib.Path
-    ) -> t.Optional[CommandList]:
-        """Build command to copy files/directories from specified paths to a destination directory.
-
-        This method creates commands to copy files/directories from the source paths provided in the
-        `files` parameter to the specified destination directory. If the source is a directory,
-        it copies the directory while allowing existing directories to remain intact.
+        files: t.List[CopyOperation], context: GenerationContext
+    ) -> CommandList:
+        """Build commands to copy files/directories from specified source paths
+        to an optional destination in the run directory.
 
-        :param files: An EntityFiles object containing the paths to copy, or None.
-        :param dest: The destination path to the Job's run directory.
-        :return: A CommandList containing the copy commands, or None if no files are provided.
+        :param files: A list of CopyOperation objects
+        :param context: A GenerationContext object that holds the Job's run directory
+        :return: A CommandList containing the copy commands
         """
-        if files is None:
-            return None
         cmd_list = CommandList()
-        for src in files.copy:
-            cmd = Command(
-                [
-                    sys.executable,
-                    "-m",
-                    "smartsim._core.entrypoints.file_operations",
-                    "copy",
-                    src,
-                ]
-            )
-            destination = str(dest)
-            if os.path.isdir(src):
-                base_source_name = os.path.basename(src)
-                destination = os.path.join(dest, base_source_name)
-                cmd.append(str(destination))
-                cmd.append("--dirs_exist_ok")
-            else:
-                cmd.append(str(dest))
-            cmd_list.commands.append(cmd)
+        for file in files:
+            cmd_list.append(file.format(context))
         return cmd_list
```

Comment:
Style nit if you want to be a hair more declarative
```suggestion
        return CommandList([file.format(context) for file in files])
```

Reply:
heck yeah!